builder/api/src/main/java/io/helidon/builder/api/RuntimeType.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (c) 2023 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.helidon.builder.api;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * This type is created from a specific prototype.
+ *
+ * @param <T> type of the prototype
+ */
+public interface RuntimeType<T extends Prototype> {
+    /**
+     * The prototype as it was received when creating this runtime object instance.
+     *
+     * @return prototype object used to create this instance
+     */
+    T prototype();
```

Comment:
why does this even matter?  Who would care about it an. way? Can we at least make this optional?  Seems like noise.

Reply:
This is to have access to the object that configured a runtime type.
For example if WebServer is configured by ServerConfig, this would give us access to ServerConfig from WebServer.
This is only used when we have the "trinity" of types (blueprint -> prototype -> runtime type).
Most common cases only have bluerpint -> prototype, and in this case this method is not used at all.
If we find out this is too intrusive when used in Helidon, we can think about doing this differently, seems useful for now

Reply:
Do we have a way to prevent ServerConfig from being exposed?  A service may not want to reveal its configuration for whatever reason..  It could be argued that it breaks encapsulation, etc.

Reply:
Not right now, we can add it (please file a follow up)

Reply:
Added it to M1 on https://github.com/helidon-io/helidon/issues/7023 since it will have an effect on the generated API